packages/axes/src/eventInput/EventInput.ts

```diff
@@ -112,13 +137,24 @@ export abstract class EventInput {
   }
 
   protected _isTouchEvent(event: InputEventType): event is TouchEvent {
-    return event.type.indexOf("touch") > -1;
+    return event.type && event.type.indexOf("touch") > -1;
   }
 
   protected _isValidButton(button: string, inputButton: string[]): boolean {
     return inputButton.indexOf(button) > -1;
   }
 
+  protected _isValidEvent(
+    event: InputEventType,
+    inputKey?: string[],
+    inputButton?: string[]
+  ): boolean {
+    return (
+      (!inputKey || isValidKey(event, inputKey)) &&
+      (!inputButton || this._isValidButton(this._getButton(event), inputButton))
```

Comment:
Isn't every event valid when there're no items in `inputKey` and `inputButton` in this case?

Reply:
Since array is type of object, an empty Array is conversed to true and `!inputButton` is conversed to false.
The reason to check `!inputButton` here is that `inputButton` is optional like in the case of a pinch gesture where there is no `inputButton` option.
But for what you said, it seemed nice to add the case where `inputButton` is an empty array to the unit test, so I added it.